addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,334 @@
+        copy=False,  # copied manually, see chatbot.script#copy
+        string='Only If', help='Show this step only if all of these answers have been selected.')
+    # forward-operator specifics
+    is_forward_operator_child = fields.Boolean(compute='_compute_is_forward_operator_child')
+
+    @api.depends('step_type')
+    def _compute_answer_ids(self):
+        for step in self:
+            if step.step_type != 'question_selection' or not step.answer_ids:
+                step.answer_ids = False
+
+    @api.depends('sequence')
+    def _compute_triggering_answer_ids(self):
+        for step in self:
+            if step.triggering_answer_ids:
+                update_command = [Command.unlink(answer.id) for answer in step.triggering_answer_ids
+                                  if answer.script_step_id.sequence >= step.sequence]
+                if update_command:
+                    step.triggering_answer_ids = update_command
+
+    @api.depends('sequence', 'triggering_answer_ids', 'chatbot_script_id.script_step_ids.triggering_answer_ids',
+                 'chatbot_script_id.script_step_ids.answer_ids', 'chatbot_script_id.script_step_ids.sequence')
+    def _compute_is_forward_operator_child(self):
+        all_steps_by_chatbot = {}
+        for chatbot in self.chatbot_script_id:
+            all_steps_by_chatbot[chatbot.id] = chatbot.script_step_ids.filtered(
+                lambda step: step.step_type in ['forward_operator', 'question_selection']
+            ).sorted(lambda s: s.sequence, reverse=True)
+        for step in self:
+            all_steps = all_steps_by_chatbot[step.chatbot_script_id.id].filtered(
+                lambda s: s.sequence < step.sequence
+            )
+            parent = step
+            while True:
+                parent = parent._get_forward_operator_parent(all_steps)
+                if not parent or parent.step_type == 'forward_operator':
+                    break
+            step.is_forward_operator_child = parent and parent.step_type == 'forward_operator'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ Ensure we correctly assign sequences when creating steps.
+        Indeed, sequences are very important within the script, and will break the whole flow if
+        not correctly defined.
+
+        This override will group created steps by chatbot_id and increment the sequence accordingly.
+        It will also look for an existing step for that chatbot and resume from the highest sequence.
+
+        This cannot be done in a default_value for the sequence field as we cannot search by
+        runbot_id.
+        It is also safer and more efficient to do it here (we can batch everything).
+
+        It is still possible to manually pass the 'sequence' in the values, which will take priority. """
+
+        vals_by_chatbot_id = {}
+        for vals in vals_list:
+            chatbot_id = vals.get('chatbot_script_id')
```

Comment:
As it is required we could try to get rid of the if ... maybe to be complete we could take chatbot_script_id or call default_get(['chatbot_script_id']) if not given ?